### PR TITLE
chore: restructure xhr/formdata tests

### PR DESCRIPTION
**Pull request by lucacasonato**
This commit restrucutres xhr/formdata tests that do not require DOM
to be .any.js tests, and thus runnable outside the browser, and in
Workers (like in Deno). This commit only moves tests around (no changes
or removals), with the exception of the "Passing a String object to
FormData.set should work" and "Passing a String object to
FormData.append should work." tests because the tests are exact replicas
of the "formdata with string" test in xhr/formdata.html. Because of that
they have been removed.

### CODE REVIEW

File: xhr/formdata.html [hunk 1]
```diff
@@ -38,16 +38,23 @@
     return fd;
   }
 
-  test(() => {
```
Comment (lucacasonato) [inline xhr/formdata.html:41]: Moved to `formdata/constructor.any.js`

File: xhr/formdata.html [hunk 2]
```diff
@@ -38,16 +38,23 @@
...
   do_test("empty formdata", new FormData(), '\n');
   do_test("formdata with string", create_formdata(['key', 'value']), 'key=value,\n');
   do_test("formdata with named string", create_formdata(['key', new Blob(['value'], {type: 'text/plain'}), 'kv.txt']), '\nkey=kv.txt:text/plain:5,');
   do_test("formdata from form", new FormData(document.getElementById('form')), 'key=value,\n');
 
+  do_test("formdata with blob", create_formdata(['key', new Blob(['value'], {type: 'text/x-value'})]), '\nkey=blob:text/x-value:5,');
+  do_test("formdata with named blob", create_formdata(['key', new Blob(['value'], {type: 'text/x-value'}), 'blob.txt']), '\nkey=blob.txt:text/x-value:5,');
+
+  // If 3rd argument is given and 2nd is not a Blob, formdata.append() should throw
+  var test = async_test('formdata.append() should throw if value is string and file name is given'); // needs to be async just because the others above are
```
Comment (annevk) [inline xhr/formdata.html:50]: const?

File: xhr/formdata/append-domform.html [hunk 3]
```diff
@@ -0,0 +1,52 @@
+<!doctype html>
+<meta charset="utf-8">
+<title>FormData.append (with dom form)
```
Comment (annevk) [inline xhr/formdata/append-domform.html:3]: with form element*

"dom form" is not terminology we tend to use.

File: xhr/formdata/append-domform.html [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+</title>
+<link rel="help" href="https://xhr.spec.whatwg.org/#dom-formdata-append">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<div id="log"></div>
+<form id="form" />
```
Comment (annevk) [inline xhr/formdata/append-domform.html:9]: This should have an explicit closing tag. Maybe fix that while here?

File: xhr/formdata/constructor-domform.html [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <!DOCTYPE html>
-<title>FormData construction from a form element</title>
+<title>FormData: constructor (with dom form)</title>
```
Comment (annevk) [inline xhr/formdata/constructor-domform.html:2]: The older title is better (as per earlier comment).

File: xhr/formdata/delete-domform.html [hunk 6]
```diff
@@ -1,7 +1,7 @@
 <!doctype html>
 <html lang=en>
 <meta charset=utf-8>
-<title>FormData: delete</title>
+<title>FormData: delete (with dom form)</title>
```
Comment (annevk) [inline xhr/formdata/delete-domform.html:4]: with form element*

File: xhr/formdata/get-from-domform.html [hunk 7]
```diff
@@ -1,7 +1,7 @@
 <!doctype html>
 <html lang=en>
 <meta charset=utf-8>
-<title>FormData: get and getAll</title>
+<title>FormData: get and getAll (with dom form)</title>
```
Comment (annevk) [inline xhr/formdata/get-from-domform.html:4]: with form element*

This applies throughout and also to the file names. I stopped noting this particular comment, but there's more instances.

**Review by annevk — COMMENTED** [review]
I wish GitHub would see more of these as moves.

Overall this looks good though and it might be nice to add some directory structure here.

**Review by annevk — APPROVED** [review]

**Comment by annevk** [comment]
Thanks @lucacasonato!

**Pull request merged**